src/data/multiset/basic.lean

```diff
@@ -1001,6 +1001,36 @@ lemma prod_map_mul [comm_monoid γ] {m : multiset α} {f g : α → γ} :
   prod (m.map $ λa, f a * g a) = prod (m.map f) * prod (m.map g) :=
 multiset.induction_on m (by simp) (assume a m ih, by simp [ih]; cc)
 
+@[simp, to_additive]
+lemma prod_map_inv' [comm_group γ] {m : multiset α} {f : α → γ} :
+  prod (m.map $ λ a, (f a)⁻¹) = (prod (m.map f))⁻¹ :=
+multiset.induction_on m (by simp_rw [map_zero, prod_zero, one_inv])
+  (λ a m ih, by simp_rw [map_cons, prod_cons, ih, mul_inv])
+
+@[simp, to_additive]
+lemma prod_map_div [comm_group γ] {m : multiset α} {f g : α → γ} :
+  prod (m.map $ λ a, f a / g a) = prod (m.map f) / prod (m.map g) :=
+multiset.induction_on m (by simp_rw [map_zero, prod_zero, div_one'])
+  (λ a m ih, by simp_rw [map_cons, prod_cons, ih, div_mul_comm])
+
+@[to_additive sum_map_nsmul]
+lemma prod_map_pow [comm_monoid γ] {m : multiset α} {f : α → γ} {n : ℕ} :
+  prod (m.map $ λ a, f a ^ n) = prod (m.map f) ^ n :=
+multiset.induction_on m (by simp_rw [map_zero, prod_zero, one_pow])
+  (λ a m ih, by simp_rw [map_cons, prod_cons, ih, mul_pow])
+
+@[to_additive]
+lemma prod_map_zpow [comm_group γ] {m : multiset α} {f : α → γ} {n : ℤ} :
+  prod (m.map $ λ a, f a ^ n) = prod (m.map f) ^ n :=
+multiset.induction_on m (by simp_rw [map_zero, prod_zero, one_zpow])
+  (λ a m ih, by simp_rw [map_cons, prod_cons, ih, mul_zpow])
+
+@[to_additive]
+lemma prod_map_zpow₀ [comm_group_with_zero γ] {m : multiset α} {f : α → γ} {n : ℤ} :
+  prod (m.map $ λ a, f a ^ n) = prod (m.map f) ^ n :=
+multiset.induction_on m (by simp_rw [map_zero, prod_zero, one_zpow₀])
```

Comment:
This and the previous are all the composition of `f` with a monoid hom, can't this be stated to generalise all of the new lemmas?

Reply:
Is `monoid_hom.map_multiset_prod` what you're looking for?

Reply:
Yeah, can't you prove these using that?

Reply:
How so? I have an outside `prod` on both the LHS and RHS.

Reply:
No you don't? eg 
```
lemma prod_map_pow [comm_group γ] {m : multiset α} {f : α → γ} {n : ℕ} :
  multiset.prod (m.map $ λ a, f a ^ n) = multiset.prod (m.map f) ^ n :=
begin
  convert (m.map f).prod_hom (pow_monoid_hom n),
  rw multiset.map_map,
  refl,
end
```
works. As far as I can tell, all your lemmas are just special cases of `multiset.prod_hom`.

Reply:
Okay, all this is nice, but it doesn't apply to `*` and `/` directly because they are binary. Any way out?

Reply:
You could make `prod_hom2`!

Reply:
Aaaargh, the rabbit hoooole

Reply:
Donr